Muhire-Josue/freelancerz
# feature(access github user)

#### What does this PR do?

- It allows accessing a user's GitHub data

#### Description of Task to be completed?

- By providing the username, it should return GitHub user data that will be used for profile

#### How should this be manually tested?

- Clone: `git clone https://github.com/Muhire-Josue/freelancerz.git`

- Install dependencies: `npm install`

- Check into this branch: `ft-access-github-user-data-#173329691`

- Run the server: `npm run dev` ==> should get this message in the terminal `Listening on port 3000`

- Hit this URL in postman to disable email notifications `PUT: http://localhost:3000/api/user/profile`

- Send this data
```
{
     "username": Muhire-Josue
}
```

#### Any background context you want to provide?

- make sure you have [node v10 - v13](https://nodejs.org/en/download/) not node v14

- make sure you have [Postgres](https://www.postgresql.org/download/) up and running

- create .env file and use the .env.example file as a reference to create the variable environment needed for this project
#### What are the relevant pivotal tracker stories?

- [173329691](https://www.pivotaltracker.com/story/show/173329691)

#### Screenshots (if appropriate)
![github_data](https://user-images.githubusercontent.com/37122177/88564193-32a7f900-d033-11ea-96d7-ea031f6a50a2.PNG)
